shell/platform/darwin/macos/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -51,6 +51,143 @@
 static NSString* const kTextKey = @"text";
 static NSString* const kTransformKey = @"transform";
 
+static NSString* const kDeleteCharacterIntent = @"TextInputClient.DeleteCharacterIntent";
+static NSString* const kDeleteToNextWordBoundaryIntent =
+    @"TextInputClient.DeleteToNextWordBoundaryIntent";
+static NSString* const kDeleteToLineBreakIntent = @"TextInputClient.DeleteToLineBreakIntent";
+static NSString* const kExtendSelectionByCharacterIntent =
+    @"TextInputClient.ExtendSelectionByCharacterIntent";
+static NSString* const kExtendSelectionToNextWordBoundaryIntent =
+    @"TextInputClient.ExtendSelectionToNextWordBoundaryIntent";
+static NSString* const kExtendSelectionToNextWordBoundaryOrCaretLocationIntent =
+    @"TextInputClient.ExtendSelectionToNextWordBoundaryOrCaretLocationIntent";
+static NSString* const kExtendSelectionToLineBreakIntent =
+    @"TextInputClient.ExtendSelectionToLineBreakIntent";
+static NSString* const kExtendSelectionVerticallyToAdjacentLineIntent =
+    @"TextInputClient.ExtendSelectionVerticallyToAdjacentLineIntent";
+static NSString* const kExtendSelectionToDocumentBoundaryIntent =
+    @"TextInputClient.ExtendSelectionToDocumentBoundaryIntent";
+static NSString* const kPerformPrivateCommandIntent =
+    @"TextInputClient.PerformPrivateTextInputCommandIntent";
+
+@interface SelectorIntent : NSObject
+@property(nonatomic, readonly) SEL selector;
+@property(nonatomic, readonly) NSArray* methodCall;
+
+- (instancetype)initWithSel:(SEL)selector methodCall:(NSArray<NSObject*>*)methodCall;
+@end
+
+@implementation SelectorIntent
+
+- (instancetype)initWithSel:(SEL)selector methodCall:(NSArray<NSObject*>*)methodCall {
+  self = [super init];
+  if (self) {
+    _selector = selector;
+    _methodCall = methodCall;
+  }
+  return self;
+}
+
++ (instancetype)intentWith:(SEL)selector methodCall:(NSArray<NSObject*>*)methodCall {
+  return [[SelectorIntent alloc] initWithSel:selector methodCall:methodCall];
+}
+
+@end
+
+static NSArray<SelectorIntent*>* const kIntentLookUp = @[
```

Comment:
Are these basically all of the [Mac keybinding responses](https://developer.apple.com/documentation/appkit/nsstandardkeybindingresponding) that Flutter also supports?

Reply:
Yes. But `selectAll` seems to be dispatched by the application menu: 

<img width="429" alt="Screen Shot 2021-12-13 at 10 07 23 AM" src="https://user-images.githubusercontent.com/31859944/145865314-b8f2f283-db1f-4356-9787-57becf76ef4f.png">

Reply:
Yes except a few that are currently don't have a framework counterpart, or it not getting dispatched via `doCommandBySelector` (e.g., selectAll).